controller/controller_server/csi_controller_server.py

```diff
@@ -387,9 +387,34 @@ def ControllerUnpublishVolume(self, request, context):
 
     def ValidateVolumeCapabilities(self, request, context):
         logger.info("ValidateVolumeCapabilities")
-        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
-        logger.info("finished ValidateVolumeCapabilities")
-        return csi_pb2.ValidateVolumeCapabilitiesResponse()
+        try:
+
+            utils.validate_validate_volume_capabilities_request(request)
+
+            secrets = request.secrets
+            user, password, array_addresses = utils.get_array_connection_info_from_secret(secrets)
+
+            array_type, volume_id = utils.get_volume_id_info(request.volume_id)
+
+            with get_agent(user, password, array_addresses, array_type).get_mediator() as array_mediator:
+                volume = array_mediator.get_object_by_id(object_id=volume_id, object_type=config.VOLUME_TYPE_NAME)
+
+            if request.volume_context:
+                utils.validate_volume_context_match_volume(request.volume_context, volume)
+            logger.info("finished ValidateVolumeCapabilities")
+            return utils.generate_csi_validate_volume_capabilities_response(request.volume_context,
+                                                                            request.volume_capabilities,
```

Comment:
according to the [spec](https://github.com/container-storage-interface/spec/blob/v1.3.0/spec.md#validatevolumecapabilities):
```
NOTE: Older plugins will parse but likely not "process" newer fields that MAY be present in capability-validation messages (and sub-messages) sent by a CO that is communicating using a newer, backwards-compatible version of the CSI protobufs. Therefore, the CO SHALL reconcile successful capability-validation responses by comparing the validated capabilities with those that it had originally requested.
```
so we should only include the capabilities we currently support, and not the capabilities we dynamically receive from the request, to avoid false claims for support of (future) "missed" values.

Reply:
done

Reply:
I don't see any change here

Reply:
its not there, its inside 'generate'

Reply:
but `generate` doesn't necessarily even know the volume.
what happens if e.g. the CO requests `ext4` but the volume has `xfs`? would it always be ok for us to re-format it later on?

Reply:
how can we know in the `controller` if the volume has `ext4 `or `xfs`?

Reply:
I don't think we currently manage this info, but that doesn't answer my question.
let's simply the scenario - in the current PR code - would we confirm to support e.g. `ext3`?
and if not, do we have a unit test to verify a scenario as such?

Reply:
[here](https://github.com/IBM/ibm-block-csi-driver/blob/a9950ee1c66276876951b1aa629725ef34ca7cc3/controller/tests/controller_server/csi_controller_server_test.py#L1507)